runtime/context.go

```diff
@@ -64,6 +86,39 @@ func GetRequestEndpointFromCtx(ctx context.Context) string {
 	return ""
 }
 
+// WithEndpointRequestHeaderField adds the endpoint request header information in the
+// request context.
+func WithEndpointRequestHeaderField(ctx context.Context, key string, value string) context.Context {
+	headers := GetEndpointRequestHeadersFromCtx(ctx)
+	headers[key] = value
+
+	return context.WithValue(ctx, endpointRequestHeader, headers)
+}
+
+// WithEndpointRequestHeadersField adds the endpoint request header information in the
+// request context.
+func WithEndpointRequestHeadersField(ctx context.Context, requestHeaders map[string]string) context.Context {
```

Comment:
This function is unused, you are only using the single header version `WithEndpointRequestHeaderField`. You should only have one. I prefer this one since it can be used for single headers as well by passing in a map with a single item.